engine/consensus/dkg/reactor_engine.go

```diff
@@ -285,7 +285,12 @@ func (e *ReactorEngine) handleEpochCommittedPhaseStarted(currentEpochCounter uin
 		return
 	}
 	if currentState != flow.DKGStateCompleted {
-		log.Warn().Msgf("checking beacon key consistency: exiting because dkg didn't reach completed state: %s", currentState.String())
+		if currentState == flow.RandomBeaconKeyCommitted {
+			// this can happen if a healthy node which succeeded the DKG restarts in the EpochCommit phase
+			log.Debug().Msg("checking beacon key consistency after EpochCommit: observed committed beacon key for most recent dkg - exiting")
+		} else {
+			log.Warn().Msgf("checking beacon key consistency after EpochCommit: exiting because dkg didn't reach completed state: %s", currentState.String())
+		}
 		return
 	}
 	snapshot := e.State.AtBlockID(firstBlock.ID())
```

Comment:
I would suggest to move this right before [line 302](https://github.com/onflow/flow-go/blob/a188c37ca0e1777dbced8908de026b8c36bf9682/engine/consensus/dkg/reactor_engine.go#L302) https://github.com/onflow/flow-go/blob/a188c37ca0e1777dbced8908de026b8c36bf9682/engine/consensus/dkg/reactor_engine.go#L302

In my head, querying the snapshot belongs to the code block around [line 302](https://github.com/onflow/flow-go/blob/a188c37ca0e1777dbced8908de026b8c36bf9682/engine/consensus/dkg/reactor_engine.go#L302) much more than it belongs to the exit condition ... just personal preference - feel free to ignore.